src/lib/buySellHelper/buySellHelper.ts

```diff
@@ -0,0 +1,84 @@
+import { TradesDataType, TRADE_STATUS } from "@jaws/db/tradesMeta";
+import { isToday } from "@jaws/lib/helpers";
+import { BuySellConstants, DEFAULT_BUY_SELL_CONFIG } from "./buySellConstants";
+
+export const getBuySellHelpers = (config?: Partial<BuySellConstants>) => {
+  const _config = { ...DEFAULT_BUY_SELL_CONFIG, ...config };
+
+  return {
+    get config() {
+      return { ..._config } as const;
+    },
+
+    /**
+     * Returns the maximum amount of money (total value) the position is
+     * allowed to drop before triggering Stop loss (1)
+     */
+    getStopLossMaxAmount: (totalAssets: number): number =>
+      totalAssets * _config.STOP_LOSS_1_PORTFOLIO_PERCENTAGE,
+
+    determineTradeStatus: (opts: {
+      trade: TradesDataType;
+      lastTradePrice: number;
+      movingAvg: number;
+      stopLossMaxAmount: number;
+    }): TRADE_STATUS | undefined => {
+      const stopLossType = determineStopLossType(opts);
+
+      if (stopLossType) {
+        return stopLossType;
+      } else if (isTakePartialProfit(opts)) {
+        return TRADE_STATUS.TAKE_PARTIAL_PROFIT;
+      }
+
+      return;
+    },
+  };
+
+  function determineStopLossType({
+    trade,
+    stopLossMaxAmount,
+    lastTradePrice,
+    movingAvg,
+  }: {
+    trade: TradesDataType;
+    stopLossMaxAmount: number;
+    lastTradePrice: number;
+    movingAvg: number;
+  }): TRADE_STATUS | undefined {
+    // Stop loss case (1)
+    if ((trade.price - lastTradePrice) * trade.quantity >= stopLossMaxAmount)
+      return TRADE_STATUS.STOP_LOSS_1;
+
+    if (!isToday(trade.created)) {
+      // Stop loss case (2)
+      if (lastTradePrice <= trade.price) return TRADE_STATUS.STOP_LOSS_2;
+
+      // Stop loss case (3) Take profit
+      if (movingAvg && lastTradePrice <= movingAvg)
+        return TRADE_STATUS.STOP_LOSS_3;
+    }
+
+    return;
+  }
+
+  /** After 10% increase in value, we take profit */
```

Comment:
1. maybe change from *is*TakePartialProfit to *should*TakePartialProfit. It just sounds nicer to me
2. This comment says 10% but I don't see that here in the code, which is a bit confusing and makes it easier for the comment to get stale since we don't directly see below if the 10% changes to 11% for example.

Reply:
I agree!